NuRadioReco/modules/io/coreas/readCoREASShower.py

```diff
@@ -0,0 +1,137 @@
+import NuRadioReco.framework.event
+import NuRadioReco.framework.station
+from NuRadioReco.framework.parameters import showerParameters as shp
+from NuRadioReco.modules.io.coreas import coreas
+from NuRadioReco.utilities import units
+from radiotools import coordinatesystems
+
+import h5py
+import numpy as np
+import time
+import re
+import os
+
+import logging
+logger = logging.getLogger('readCoREASShower')
+
+
+class readCoREASShower:
+
+    def __init__(self):
+        self.__t = 0
+        self.__t_event_structure = 0
+        self.__t_per_event = 0
+
+    def begin(self, input_files, det = None, logger_level=logging.NOTSET):
+        """
+        begin method
+
+        initialize readCoREASShower module
+
+        Parameters
+        ----------
+        input_files: input files
+            list of coreas hdf5 files
+        det: genericDetector object
+            If a genericDetector is passed, the stations from the CoREAS file
+            will be added to it and the run method returns both the event and
+            the detector
+        """
+        self.__input_files = input_files
+        self.__current_input_file = 0
+        self.__det = det
+        logger.setLevel(logger_level)
+
+
+    def run(self):
+        """
+        read in a full CoREAS simulation
+
+        """
+        while (self.__current_input_file < len(self.__input_files)):
+            t = time.time()
+            t_per_event = time.time()
+
+            filesize = os.path.getsize(self.__input_files[self.__current_input_file])
+            if(filesize < 18456 * 2):
+                logger.warning("file {} seems to be corrupt, skipping to next file".format(
+                       self.__input_files[self.__current_input_file]))
+                self.__current_input_file += 1
+                continue
+
+            logger.info('Reading %s ...' % self.__input_files[self.__current_input_file])
+
+            corsika = h5py.File(self.__input_files[self.__current_input_file], "r")
+            logger.info("using coreas simulation {} with E={:2g} theta = {:.0f}".format(
+                self.__input_files[self.__current_input_file], corsika['inputs'].attrs["ERANGE"][0] * units.GeV,
+                corsika['inputs'].attrs["THETAP"][0]))
+
+            f_coreas = corsika["CoREAS"]
+
+            evt = NuRadioReco.framework.event.Event(corsika['inputs'].attrs['RUNNR'], corsika['inputs'].attrs['EVTNR'])
```

Comment:
Is there any reason to not just iterate the event ids? The problem with the way it is right now is that for example in our simulations each event has the same id and run number, which causes problems. Are the ids in the CoREAS files actually important in practice?

Reply:
Hi, they can be important. For a library with discrete bins in arrival direction and energy we used the run number as an identifier. 
The event number we do not use (but i would not be sure if that is the case for every one).

Reply:
That could be an issue then. In the simulations we have, they all have the same ID.

Reply:
One could have an additional option overwriting them, but this is somewhat messy

Reply:
Maybe we could store it in the SimShower. That seems like a natural place to put information like this.
It's probably a good idea to have a place to put some more information about the simulation itself. That would also be useful for neutrinos, for stuff like which ice model was used

Reply:
We should definitely store some more data on event level such as used atmospheric model, ...
I am still not sure if we should store something like the run_number on simshower lvl. I think this belongs to the event lvl.

Reply:
Yes, that is not ideal. But always taking the ID from the CoREAS causes problems as well. It should be at least an option to iterate the ID instead of taking them from the file.
Can you take care of that? I want to get this PR done.

Reply:
This is not yet resolved right? I think we can't have events without a unique run/eventid pair.

Reply:
Hi, 
I added a simple option to set a ascending run and event number. Are you fine with that?

Reply:
Have you tested it? It looks weird. You use the parameter set_ascending_run_and_event_number in the run function, event though it is a parameter of the begin function.
Also, please explain what the parameter does in the docstring

Reply:
Hi,
have not tested it. I can try to do this next week. but it should be straight forward. I can also add it to the docstring.
Felix

Reply:
Hi, 
puh that was very stupid.. sorry for that. It should work now. I tested it a bit and I hope it is fine now. The solution might be a bit unorthodox to use the value to set the run number and event id  also as bool but it saves one variable.
I also added some more documentation.
Cheers